src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1521,6 +1556,81 @@ public bool MoveNext()
             return true;
         }
     }
+
+    /// <summary>
+    /// The simple enumerator class is used for the range operator '..'
+    /// in expressions like 'A'..'B' | ForEach-Object { $_ }
+    /// </summary>
+    internal class CharRangeEnumerator : IEnumerator
```

Comment:
Should this implement `IEnumerable<String>` instead?

Reply:
I think we cast all to `object` type before send to pipe. So it seems there is no sense to specify here the exact type.